src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -125,7 +330,7 @@ public void Abort(ConnectionAbortedException error)
         }
     }
 
-    public ValueTask<FlushResult> FlushAsync(IHttpOutputAborter? outputAborter, CancellationToken cancellationToken)
+    private ValueTask<FlushResult> FlushAsync(Http2Stream stream, bool writeHeaders, IHttpOutputAborter? outputAborter, CancellationToken cancellationToken)
     {
         lock (_writeLock)
```

Comment:
This is the lock that was causing contention problems. Can it be removed?

Reply:
Not fully. There will be less contention but window updates fire out of band from the normal processing so it's still required for that. Also writing 100 continue and probably some other obscure things.